### PR TITLE
Fixed AY import to keep caches consistent when doing so several times in a row

### PR DESCRIPTION
**Pull request by mchf**
bnc#866742

### CODE REVIEW

File: src/modules/Lan.rb [hunk 1]
```diff
@@ -666,34 +666,14 @@ def WriteOnly
     # @param [Hash] settings settings to be imported
     # @return true on success
     def Import(settings)
-      settings = deep_copy(settings)
-      NetworkInterfaces.Import("netcard", Ops.get_map(settings, "devices", {}))
-      Builtins.foreach(NetworkInterfaces.List("netcard")) do |device|
-        LanItems.AddNew
-        Ops.set(LanItems.Items, LanItems.current, { "ifcfg" => device })
-      end
+      settings = {} if settings.nil?
```
Comment (jreidinger) [inline src/modules/Lan.rb]: `settings ||= {}`

File: src/modules/Lan.rb [hunk 2]
```diff
@@ -703,11 +683,9 @@ def Import(settings)
       else
         NetworkService.use_wicked
       end
-      if Builtins.haskey(settings, "ipv6")
-        @ipv6 = Ops.get_boolean(settings, "ipv6", true)
-      end
 
-      LanItems.modified = true
+      @ipv6 = settings["ipv6"] || true
```
Comment (jreidinger) [inline src/modules/Lan.rb]: it is a bit confusing to have spread over whole file settings default. What about set it at beginning or maybe even better use merging? like `settings = DEFAULT_SETTINGS.merge(settings)`

Comment (mchf) [inline src/modules/Lan.rb]: well, it is. And not only in one file. But I'm not able to save them all now, so I'm only touching those directly affected by a PR.

File: src/modules/LanItems.rb [hunk 3]
```diff
@@ -1088,6 +1081,45 @@ def Read
...
+    def reset_cache
+      LanItems.Items = {}
+
+      @modified = false
+    end
+
+    # Imports data from AY profile
+    #
+    # As network related configuration is spread over the whole AY profile
+    # the function requires hash map with whole AY profile.
```
Comment (mvidner) [inline src/modules/LanItems.rb]: Not quite. It is not the whole profile but rather its `<profile>/<networking>` element, AFTER it has been processed by [FromAY](https://github.com/yast/yast-network/blob/6ca4d37d55fd1517751cef86b696e9692dedbd53/src/clients/lan_auto.rb#L169).

File: src/modules/Lan.rb (hunk 2 above)
Comment (mvidner) [inline src/modules/Lan.rb]: `@ipv6 = settings["ipv6"] || true` NOOO!

That is the classical case where we must not use the `|| default` for supplying the default value. What if the user explicitly had `settings["ipv6"] = false`? That's why we must use `settings.fetch("ipv6", true)`.

File: src/modules/LanItems.rb (hunk 3 above)
Comment (mchf) [inline src/modules/LanItems.rb]: right. Bad wording. I'm going to rephrase it.

File: src/modules/Lan.rb (hunk 2 above)
Comment (mchf) [inline src/modules/Lan.rb]: thanks

**Comment by mvidner** [comment]
Just that one `fetch` thing, otherwise it looks good.

**Comment by mvidner** [comment]
LGTM

**Comment by mchf** [comment]
Thanks for reviews

**Pull request merged**